File: nhglib-core/src/io/github/movementspeed/nhglib/data/models/serialization/components/UiComponentJson.java
package io.github.movementspeed.nhglib.data.models.serialization.components;

import com.badlogic.gdx.utils.JsonValue;
import io.github.movementspeed.nhglib.assets.Asset;
import io.github.movementspeed.nhglib.data.models.serialization.AssetJson;
import io.github.movementspeed.nhglib.data.models.serialization.ComponentJson;
import io.github.movementspeed.nhglib.runtime.ecs.components.graphics.UiComponent;

public class UiComponentJson extends ComponentJson {
    @Override
    public void parse(JsonValue jsonValue) {
        UiComponent uiComponent = nhg.entities.createComponent(entity, UiComponent.class);
        uiComponent.fileName = jsonValue.getString("fileName", "");

        JsonValue dependenciesJson = jsonValue.get("dependencies");

        for (int i = 0; i < dependenciesJson.size; i++) {
            AssetJson assetJson = new AssetJson();
            assetJson.parse(dependenciesJson.get(i));

            Asset dependency = assetJson.get();
            uiComponent.dependencies.add(dependency);
        }

        output = uiComponent;
    }
}

File: nhglib-core/src/io/github/movementspeed/nhglib/data/models/serialization/components/UiComponentJson.kt
package io.github.movementspeed.nhglib.data.models.serialization.components

import com.badlogic.gdx.utils.JsonValue
import io.github.movementspeed.nhglib.assets.Asset
import io.github.movementspeed.nhglib.core.ecs.components.graphics.UiComponent
import io.github.movementspeed.nhglib.data.models.serialization.AssetJson
import io.github.movementspeed.nhglib.data.models.serialization.ComponentJson

class UiComponentJson : ComponentJson() {
    override fun parse(jsonValue: JsonValue) {
        val uiComponent = nhg!!.entities.createComponent(entity, UiComponent::class.java)
        uiComponent.fileName = jsonValue.getString("fileName", "")

        val uiType = jsonValue.getString("uiType", "screen").toLowerCase()

        if (uiType.contentEquals("screen")) {
            uiComponent.type = UiComponent.Type.SCREEN
        } else if (uiType.contentEquals("panel")) {
            uiComponent.type = UiComponent.Type.PANEL
        }

        val dependenciesJson = jsonValue.get("dependencies")

        for (i in 0 until dependenciesJson.size) {
            val assetJson = AssetJson()
            assetJson.parse(dependenciesJson.get(i))

            val dependency = assetJson.get()
            uiComponent.dependencies.add(dependency)
        }

        val actorNamesJson = jsonValue.get("actors")

        for (i in 0 until actorNamesJson.size) {
            val actor = actorNamesJson.getString(i)
            uiComponent.actorNames.add(actor)
        }

        output = uiComponent
    }
}